File: nexus-backend/nexus-tasks/src/main/java/com/projectnexus/tasks/domain/Task.java
package com.projectnexus.tasks.domain;

import com.projectnexus.common.entity.BaseTenantEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.UUID;

/**
 * Cross-team task that can be linked to a Data Contract, Payload, or Deviation.
 * Supports Kanban-style workflow management.
 */
@Entity
@Table(name = "tasks")
@Getter
@Setter
@NoArgsConstructor
public class Task extends BaseTenantEntity {

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TaskStatus status = TaskStatus.TODO;

    @Column(name = "assignee_id")
    private UUID assigneeId;

    @Column(name = "due_date")
    private LocalDateTime dueDate;

    @Enumerated(EnumType.STRING)
    @Column(name = "linked_to_type")
    private LinkType linkedToType;

    @Column(name = "linked_to_id")
    private UUID linkedToId;

    @Column(name = "ai_generated", nullable = false)
    private boolean aiGenerated = false;
}

File: nexus-frontend/src/types/task.ts
export type TaskStatus = 'TODO' | 'IN_PROGRESS' | 'IN_REVIEW' | 'DONE';
export type LinkedEntityType = 'CONTRACT' | 'PAYLOAD' | 'DEVIATION';

export interface Task {
  id: string;
  tenantId: string;
  title: string;
  description: string;
  status: TaskStatus;
  assigneeId: string | null;
  dueDate: string | null;
  linkedToType: LinkedEntityType | null;
  linkedToId: string | null;
  aiGenerated: boolean;
  createdAt: string;
  updatedAt: string;
  createdBy: string;
}

export interface CreateTaskRequest {
  title: string;
  description?: string;
  assigneeId?: string;
  dueDate?: string;
  linkedToType?: LinkedEntityType;
  linkedToId?: string;
}
